esti/hooks_test.go

```diff
@@ -18,7 +19,18 @@ import (
 )
 
 //go:embed action_files/*.yaml
-var actions embed.FS
+var actionsPath embed.FS
+
+var (
+	resultVal = make([]*webhookEventInfo, 0)
+	mu        sync.RWMutex
+)
```

Comment:
You just added an `esti` package level mutex. Can you create a simple struct for it to avoid misuse?